migrator.go

```diff
@@ -0,0 +1,84 @@
+package influxdb
+
+import "context"
+
+// MigrationManager is the manager of migrators.
+type MigrationManager struct {
+	Migrators []Migrator
+}
+
+// Migrate will perform the migration.
+func (mg MigrationManager) Migrate(ctx context.Context, Up bool, errCh chan<- error) {
+	for _, m := range mg.Migrators {
+		if !m.NeedMigration() {
+			return
+		}
+		bucket := m.Bucket()
+		srcs, err := m.ScanBytes(ctx, bucket)
+		if err != nil {
+			errCh <- &Error{
+				Err: err,
+				Op:  "Data Migration Scan",
+			}
+			continue
+		}
+		for key, src := range srcs {
+			var dst []byte
+			if Up {
+				dst, err = m.Up(src)
+			} else {
+				dst, err = m.Down(src)
+			}
+			if err != nil {
+				errCh <- err
+				continue
+			}
+			if err = m.StoreBytes(ctx, bucket, key, dst); err != nil {
+				errCh <- err
+			}
+		}
+	}
+	// close it
+	errCh <- nil
+}
+
+// Migrator is the interface of schema migration.
+type Migrator interface {
+	MigragorService
```

Comment:
So if I understand correctly all migrations have to implement the migration service? This introduces a lot of boilerplate on each request and gives no advantage as far as i can see.